R/observations.R

```diff
@@ -60,6 +62,12 @@ observation_impute <- function(W_kernel, S, x_train, x_test, w_threshold = .7, n
   )
   dt_melt[, index_s := nms_vec[id_combination]]
 
+  # Extracts zero and full conditional to be handled separately
```

Comment:
In terms of these changes I would actually like to see that it is an improvement in speed. It think it actually might be slower. Pushing two additional rows to c++ shouldn't mean anything?

Reply:
As I commented below it is about generating 2*n_samples rows, not just two. It could be changed in the c++ code, but it was much easier to do it outside.

Reply:
Sure. But the percentage these two combinations represents in this case is `2 / total number of combinations`? I don't see the incentive to introduce additional complexity and book-keeping here. This was designed this way to avoid copying the data.

Reply:
Yes. The important thing is that the output has one row with the test data and weight 1 when conditioning on no features, and another copy of that when conditioning on all features. There should be no other rows with those combinations, and they should only be present if passed to `prepare_data`. So, if you think it is better to replace the rows after passing them to `observation_impute_cpp`, I could try to do that instead.

Reply:
@nikolase90 What do you think about my suggestion here? Let me know if it is unclear what I mean.